lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -3,6 +3,190 @@
+     * @param {@link (AuthenticationParameters:type)}
+     */
+    loginRedirect(request: msalAuth.AuthenticationParameters): void {
+        throw new Error("Method not implemented.");
+    }
+
+    /**
+     * Use when you want to obtain an access_token for your API by redirecting the user's browser window to the authorization endpoint.
+     * @param {@link (AuthenticationParameters:type)}
+     *
+     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+     */
+    acquireTokenRedirect(request: msalAuth.AuthenticationParameters): void {
+        throw new Error("Method not implemented.");
+    }
+
+    // #endregion
+
+    // #region Popup Flow 
+
+    /**
+     * Use when initiating the login process via opening a popup window in the user's browser
+     *
+     * @param {@link (AuthenticationParameters:type)}
+     *
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
+    loginPopup(request: msalAuth.AuthenticationParameters): Promise<msalAuth.TokenResponse> {
+        throw new Error("Method not implemented.");
+    }
+
+    /**
+     * Use when you want to obtain an access_token for your API via opening a popup window in the user's browser
+     * @param {@link AuthenticationParameters}
+     *
+     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     */
+    acquireTokenPopup(request: msalAuth.AuthenticationParameters): Promise<msalAuth.TokenResponse> {
+        throw new Error("Method not implemented.");
+    }
+
+    // #region Silent Flow
+
+    /**
+     * Use this function to obtain a token before every call to the API / resource provider
+     *
+     * MSAL return's a cached token when available
+     * Or it send's a request to the STS to obtain a new token using a hidden iframe.
+     *
+     * @param {@link AuthenticationParameters}
+     *
+     * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+     * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+     *
+     */
+    acquireTokenSilent(request: msalAuth.AuthenticationParameters): Promise<msalAuth.TokenResponse> {
+        throw new Error("Method not implemented."); 
+    }
+
+    // #endregion
+
+    // #region Window Navigation
+
+    /**
+     * @hidden
+     * Used to redirect the browser to the STS authorization endpoint
+     * @param {string} urlNavigate - URL of the authorization endpoint
```

Comment:
is this... necessary? does the endpoint vary in this case? even if it does, we may want to enumerate the various calls